server.js

```diff
@@ -21,10 +14,75 @@ app.use(express.json());
 
 // Start defining your routes here
 app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
+  const endpoints = listEndpoints(app);
+  res.json({
+    message: "Welcome to the Harry Potter page!",
+    endpoints: endpoints
+  });
+});
+
+// Route to get all characters or filter by house, year, role
+app.get("/harryPotterCharacters", (req, res) => {
+  const house = req.query.house; // Query parameter for house
+  const year = req.query.yearIntroduced; // Query parameter for yearIntroduced
+  const role = req.query.role; // Query parameter for role
+
+  // Filter by house
+  if (house) {
+    const filteredCharacters = harryPotterCharactersData.filter(character =>
+      character.house.toLowerCase() === house.toLowerCase()
+    );
+    res.json(filteredCharacters);
+  }
+  // Filter by role
+  else if (role) {
+    const characterRole = harryPotterCharactersData.filter(character =>
+      character.role.toLowerCase() === role.toLowerCase()
+    );
+    res.json(characterRole);
+  }
+  // Filter by yearIntroduced
+  else if (year) {
+    const charactersIntroducedInYear = harryPotterCharactersData.filter(character =>
+      character.yearIntroduced === +year
+    );
+    res.json(charactersIntroducedInYear);
+  }
+  else {
+    res.json(harryPotterCharactersData);  // No filters = return all characters
+
+  }
+});
```

Comment:
Nice job making use of query params to filter ⭐ Could you think of a way to improve this code so that the user can filter on two things at once? E.g. `/characters?role=student&year=1997`